django/contrib/gis/geoip2/base.py

```diff
@@ -92,6 +92,9 @@ def __init__(self, path=None, cache=0, country=None, city=None):
             if os.path.isfile(city_db):
                 self._city = geoip2.database.Reader(city_db, mode=cache)
                 self._city_file = city_db
+
+            if not self._country and not self._city:
```

Comment:
I think this should be `if not self._reader` - that covers the same expression, but it's the call that's ultimately failing in the original bug? With this fix, if there were a change to the implementation of `_reader` the original bug would return.

Reply:
@hugorodgerbrown you are completely right. If we use `_reader` property and its implementation will change later, then there is a chance that the bug would return. Therefore, I use `_country` and `_city`.

Reply:
Agreed, but the underlying bug itself is caused by a call to `self._reader`, and not city or country. As it stands this would not solve the original ticket if a city or country database existed but for some reason the reader was null.

Reply:
Alternatively, the `__repr__` and `info` methods could be updated to handle nulls?

Reply:
Ok, I understood. I will check it asap.